apps/js_urls/serializer.py

```diff
@@ -0,0 +1,130 @@
+# Python Standard Library Imports
+import json
+from urllib.parse import urljoin
+
+# Django Imports
+from django.urls import get_resolver
+from django.urls.resolvers import (
+    RegexPattern,
+    RoutePattern,
+    URLPattern,
+    URLResolver,
+)
+
+# HTK Imports
+from htk.apps.js_urls.constants.regexp import (
+    URL_ARG_RE,
+    URL_KWARG_RE,
+    URL_OPTIONAL_CHAR_RE,
+    URL_OPTIONAL_GROUP_RE,
+    URL_PATH_RE,
+)
+from htk.apps.js_urls.utils import replace
+from htk.utils import htk_setting
+
+
+def get_urls_as_dict(resolver=None):
+    resolver = resolver or get_resolver()
+    return dict(_parse_resolver(resolver))
+
+
+def get_urls_as_json(resolver=None):
+    resolver = resolver or get_resolver()
+    return json.dumps(get_urls_as_dict(resolver))
+
+
+def _parse_resolver(
+    resolver, current_namespace=None, url_prefix=None, include_all=False
+):
+    allowed_urls = htk_setting('HTK_APPS_JS_URLS_ALLOWED')
+    namespace = (
+        f"{current_namespace}:{resolver.namespace}"
+        if current_namespace
+        else resolver.namespace
+    )
+    include_all = include_all or (namespace in allowed_urls)
+
+    urls = []
+    for url_pattern in resolver.url_patterns:
```

Comment:
Not sure what this function is doing as it doesn't have a docstring. The different sections could also use comments.

Unless performing side effects, it might be better to write it as `urls = [_some_transform_fn(url_pattern) for url_pattern in resolver.url_patterns]`

Reply:
There are two types of url items in `url_patterns` list in `urls.py`

One is normal URL with a view function defined to it like `re_path(r'^index$', views.index_page, name='index')`
Other one is including another `urls.py` with include `re_path(r'^account', include('account.urls'))`

We are traversing nested url patterns with same function.

Added descriptive inline comments for this. I think code can explain better than me 😄